prog/vnet/subnet_nexus.rb

```diff
@@ -38,13 +38,90 @@ def wait
       hop :refresh_mesh
     end
 
+    when_refresh_keys_set? do
+      ps.update(state: "refreshing_keys")
+      hop :refresh_keys
+    end
+
     nap 30
   end
 
+  def gen_encryption_key
+    "0x" + SecureRandom.bytes(36).unpack1("H*")
+  end
+
+  def gen_spi
+    "0x" + SecureRandom.bytes(4).unpack1("H*")
+  end
+
+  def gen_reqid
+    SecureRandom.random_number(100000) + 1
+  end
+
+  def refresh_keys
+    payload = {}
+
+    ps.nics.each do |nic|
+      payload[nic.id] = {
+        spi4: gen_spi,
+        spi6: gen_spi,
+        reqid: gen_reqid
+      }
+      nic.update(encryption_key: gen_encryption_key)
+    end
+
+    ps.nics.each do |nic|
+      bud Prog::Vnet::RekeyNic, {payload: payload, subject_id: nic.id}
+    end
+
+    hop :wait_inbound_setup
+  end
+
+  def wait_inbound_setup
+    reap
+    DB.transaction do
+      if strand.children.filter { |stc| stc.prog == "Vnet::RekeyNic" }.all? { |stc| stc.label == "wait_outbound_trigger" }
```

Comment:
I'm not a huge fan of this waiting around for particular labels in the programs...what's the code bloat cost of using return/exit values?

Reply:
...I suppose the problem is, they need to continue a process with the same stack frames/context right? so set-up happens *again*. 

An alternative way to do this: via semaphores. But I'm not sure if this is really any better. And it's certainly more expensive (additional database modifications)

Reply:
Dividing `Vnet::RekeyNic` to two progs them and recreating stack for them feels more natural to me. We can bud first and reap all to wait, then bud second one

Reply:
Somehow I think the line penalty will not be *so* great, so I'm sort of erring in that direction, but it's plausible that it may come out better kinda monolithic. This is not an easily predicted outcome without doing a bit of keyboard smashing...